extensions/pkg/apis/config/v1alpha1/types.go

```diff
@@ -28,7 +28,8 @@ type HealthCheckConfig struct {
 	// defaults to 30 sec
 	SyncPeriod metav1.Duration `json:"syncPeriod"`
 	// ShootRESTOptions allow overwriting certain default settings of the shoot rest.Config.
-	ShootRESTOptions RESTOptions `json:"shootRESTOptions"`
+	// +optional
+	ShootRESTOptions *RESTOptions `json:"shootRESTOptions"`
```

Comment:
As it is not optional/pointer, we have to also add the `,omitempty` part.

```suggestion
	ShootRESTOptions *RESTOptions `json:"shootRESTOptions,omitempty"`
```

Reply:
done